modules/programs/mangohud.nix

```diff
@@ -0,0 +1,100 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mangohud;
+
+  renderOption = option:
+    rec {
+      int = toString option;
+      float = int;
+      path = int;
+      bool = "false";
+      string = option;
+      list = concatStringsSep "," option;
+    }.${builtins.typeOf option};
+
+  renderLine = k: v: (if isBool v && v then k else "${k}=${renderOption v}");
+  renderSettings = attrs:
+    strings.concatStringsSep "\n" (attrsets.mapAttrsToList renderLine attrs)
+    + "\n";
+
+in {
+  options = {
+    programs.mangohud = {
+      enable = mkEnableOption "Mangohud";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.mangohud;
+        defaultText = literalExample "pkgs.mangohud";
+        description = "The Mangohud package to install.";
+      };
+
+      setEnvVars = mkOption {
```

Comment:
maybe this should warn the user in some way that it only works when home-manager manages your x-session or the .xprofiles is set appropriately?
see https://github.com/nix-community/home-manager/issues/1011#issuecomment-850889575

Reply:
Perhaps is sufficient to update the installation instructions, right now they focus mostly on sourcing `hm-session-vars.sh` when not using the shell modules. It should also say that you need to source the file in your graphical session.